README.md

```diff
@@ -87,12 +68,73 @@ Learn from experienced cooks and grocery shoppers and provide the resulting guid
 
 ## Getting Started
 
-In this phase of the project, all raw data exist in CSV's as downloaded from Instacart [as linked below](##DataLinks). In order to get up and running yourself, you will need to download these large files and move all `.csv` files into `./data/external/`
+In this phase of the project, all raw data exist in CSV's as downloaded from Instacart [as linked below](##DataLinks). In order to get up and running yourself, you will need to download these large files into `./data/external/` along with several other setup steps required before running the application.
+
+Luckily for you, all this can be easily automated using the included Makefile
+
+### Using the Makefile
+
+Makefile directives can be executed by running `make directive`, or `make VAR=X directive` if you want to set environment variable `VAR` as `X` before executing a directive. Examples will follow:
+
+Before beginning work on this project, it is recommended that you create a virtual **environment** with the required packages. Depending on your preferences, this can be done via `virtualenv` or `conda`  
+*Note:* if `make conda` fails for you, you may have to run `conda activate instacart && pip install -r requirements.txt`
+
+```bash
+make conda
+make venv
+```
+
+**Download** raw data from Instacart website and unpack in the appropriate location
+
+```bash
+make data
+```
+
+**Upload** raw data files (`data/external/*.csv`) to S3 bucket  
+*Note:* alternatively, `python src/upload_s3.py --bucket <bucket-name> --dir <local-dir> --file <local-file>` will upload any files matching `local-file` pattern within `local-dir` to S3 in the specified bucket
+
+```bash
+make s3
+make BUCKET="cool-s3-bucket" s3
+```
+
+**Generate features** from the raw data for later use in model development  
+*Warning:* this can be a compute-heavy process and may not run well (or at all) on limited resources. feature generation involves clustering on a large dataset and takes about 10 minutes to run on my MacBook with 2.9GHz i7 and 16GB RAM
+
+```bash
+make features
+```
+
+**Create database** to persist basket (order type) data  
+*Note:* the `rds` mode will only work if valid MYSQL config is available in the environment variables (e.g. MYSQL_{USER, PASSWORD, HOST, PORT})
+
+```bash
+make db
+make MODE="rds" db
+```
+
+**Ingest** the created feature data (baskets.csv)  
+*Note:* this will also create the table if it does not yet exist
+
+```bash
+make ingest
+make MODE="rds" ingest
+```
```

Comment:
Is it necessary to document every single make in the readme? Just thinking from the point of view of a new user that would probably not run each incremental step.

Reply:
probably sufficient to document `make all` and mention that the individual directives can be referenced from `Makefile` itself - will ask Chloe/Fausto for second opinion as well